src/Components/Alert/Hooks/useAlertContext.tsx

```diff
@@ -188,6 +211,48 @@ export const AlertProvider: FC<AlertProviderProps> = ({
     basis.current = state
   }, [state])
 
+  // Fetching the display name and labels for alert name input placeholder and the criteria pills, when the criteria changes.
+
+  const { debouncedValue: debouncedCriteria } = useDebouncedValue({
+    value: state.criteria,
+    delay: 200,
+    settings: { leading: true },
+  })
```

Comment:
Debouncing the preview fetch with `leading` set to `true` in order to get the result as fast as possible and prevent too many requests.